src/shared/sam/debugger/typescriptSamDebug.ts

```diff
@@ -18,6 +19,12 @@ import { DefaultSamLocalInvokeCommand, WAIT_FOR_DEBUGGER_MESSAGES } from '../cli
 import { runLambdaFunction, waitForPort } from '../localLambdaRunner'
 import { SamLaunchRequestArgs } from './awsSamDebugger'
 
+const TS_BUILD_DIR = 'aws-toolkit-ts-output'
```

Comment:
Doing this is ok but now there's going to be inconsistencies between the runtimes. I can make a PR to simplify some of the run/debug codepaths to make things consistent while still accommodating your improvements.

Reply:
Good call. Updating to use outDir in tsconfig by default (or buildDir under sam in launch configuration if set).